scripts/find-unused-fields.ts

```diff
@@ -0,0 +1,349 @@
+ */
+const ignoredFragments: Record<string, Ignored> = {
+  ArtworkActions_artwork: {
+    fields: ["dimensions", "image"],
+  },
+  ArtworkDetailsAboutTheWorkFromPartner_artwork: {
+    fields: ["partner/locations/city", "partner/profile/id"],
+  },
+  ArtworkFilter_artist: {
+    fields: [
+      "name",
+      "filtered_artworks/aggregations/counts/name",
+      "filtered_artworks/aggregations/counts/id",
+    ],
+  },
+  ArtworkImageBrowser_artwork: {
+    fields: [
+      "images/placeholder",
+      "images/placeholder",
+      "images/deepZoom/Image",
+      "images/aspectRatio",
+    ],
+  },
+}
+
+function getLanguageService(): ts.LanguageService {
+  const { options, ...others } = ts.convertCompilerOptionsFromJson(
+    json5.parse(fs.readFileSync("./tsconfig.json").toString()).compilerOptions,
+    "."
+  )
+
+  if (others.errors && others.errors.length > 0) {
+    throw new Error(JSON.stringify(others.errors))
+  }
+
+  const sourceFiles = klawSync("./src", {
+    traverseAll: true,
+    filter(item) {
+      return item.path.endsWith(".ts") || item.path.endsWith(".tsx")
+    },
+  }).map(item => item.path)
+
+  // Create the language service files
+  return ts.createLanguageService(
+    {
+      getScriptFileNames: () => sourceFiles,
+      getScriptVersion: _ => "0",
+      getScriptSnapshot: fileName => {
+        if (!fs.existsSync(fileName)) {
+          return undefined
+        }
+
+        return ts.ScriptSnapshot.fromString(
+          fs.readFileSync(fileName).toString()
+        )
+      },
+      getCurrentDirectory: () => process.cwd(),
+      getCompilationSettings: () => options,
+      getDefaultLibFileName: opts => ts.getDefaultLibFilePath(opts),
+      fileExists: ts.sys.fileExists,
+      readFile: ts.sys.readFile,
+      readDirectory: ts.sys.readDirectory,
+    },
+    ts.createDocumentRegistry()
+  )
+}
+
+const compare = (a, b) => (a < b ? -1 : a > b ? 1 : 0)
+
+function getSources(languageService: ts.LanguageService): ts.SourceFile[] {
+  let arg = process.argv[2]
+  if (arg) {
+    if (!fs.existsSync(arg)) {
+      if (fs.existsSync(`src/__generated__/${arg}.graphql.ts`)) {
+        arg = `src/__generated__/${arg}.graphql.ts`
+      } else {
+        console.error(chalk.red.bold("Bad argument"), chalk.bold("arg"))
+        console.error(
+          "Try a relay fragment name or the path of a generated relay typings file."
+        )
+        process.exit(1)
+      }
+    }
+    return [
+      languageService
+        .getProgram()
+        .getSourceFile(
+          "src/__generated__/ArtworkImageBrowser_artwork.graphql.ts"
+        ),
+    ]
+  } else {
+    // get all files
+    return languageService
+      .getProgram()
+      .getSourceFiles()
+      .filter(
+        f =>
+          f.fileName.match("__generated__") &&
+          !f.fileName.endsWith("Query.graphql.ts")
+      )
+      .sort((a, b) => compare(basename(a.fileName), basename(b.fileName)))
+  }
+}
+
+class CheckContext {
```

Comment:
As there's a lot of complex behavior in this class, mind add a comment at the top and / or comments inline?